src/compiler/crystal/macros/methods.cr

```diff
@@ -2678,6 +2686,18 @@ private def macro_raise(node, args, interpreter)
   node.raise msg, exception_type: Crystal::MacroRaiseException
 end
 
+private def macro_warning(node, args, interpreter)
+  msg = args.map do |arg|
+    arg.accept interpreter
+    interpreter.last.to_macro_id
+  end
+  msg = msg.join " "
```

Comment:
While playing with the other macro error stuff I noticed that you can actually pass a variadic amount of arguments to macro raise? This keeps that behavior for consistency, but could also do a follow up to make it an error if are 0 or more than 1 arguments passed to `#warnings` and add a deprecation to `#raise`.

```cr
{% raise "foo", "bar", 123 %}
```
```cr
$ crystal run --error-trace test.cr 
In test.cr:1:1

 1 | {% raise "foo", "bar", 123 %}
     ^
Error: foo bar 123
```